### PR TITLE
add jenkins api & update struct

### PR DESCRIPTION
**Pull request by soulseen**
Signed-off-by: soulseen <sunzhu@yunify.com>

### CODE REVIEW

**Comment by soulseen** [comment]
@runzexia @zryfish @wansir

File: pkg/apis/devops/v1alpha2/register.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"kubesphere.io/kubesphere/pkg/apiserver/devops"
 	"kubesphere.io/kubesphere/pkg/apiserver/runtime"
+	modelDevops "kubesphere.io/kubesphere/pkg/models/devops"
```
Comment (zryfish) [inline pkg/apis/devops/v1alpha2/register.go]: conventionally, use all lowercase letters for imported package name

File: pkg/apis/devops/v1alpha2/register.go [hunk 2]
```diff
@@ -87,7 +95,8 @@ func addWebService(c *restful.Container) error {
 		Param(webservice.PathParameter("pipelineName", "pipeline name")).
 		Param(webservice.PathParameter("projectName", "devops project name")).
 		Param(webservice.PathParameter("branchName", "pipeline branch name")).
-		Param(webservice.PathParameter("runId", "pipeline runs id")))
+		Param(webservice.PathParameter("runId", "pipeline runs id")).
+		Returns(200, "success", modelDevops.RunPipeline{}))
```
Comment (zryfish) [inline pkg/apis/devops/v1alpha2/register.go]: use status code `200`, `success` as constant here

File: pkg/models/devops/devops.go [hunk 3]
```diff
@@ -36,7 +36,7 @@ func init() {
 	flag.StringVar(&JenkinsUrl, "jenkins-url", "http://ks-jenkins.kubesphere-devops-system.svc.cluster.local:80", "jenkins server host")
 }
 
-func GetPipeline(projectName, pipelineName string, req *http.Request) (*Pipeline, error) {
+func GetPipeline(projectName, pipelineName string, req *http.Request) (*TypePipeline, error) {
```
Comment (zryfish) [inline pkg/models/devops/devops.go]: what does `TypePipeline` and `RunPipeline` mean? confusing here

File: pkg/models/devops/json.go [hunk 4]
```diff
@@ -17,148 +17,287 @@
...
+		} `json:"self"`
+	} `json:"_links"`
+	Avatar                      string `json:"avatar"`
+	JenkinsOrganizationPipeline bool   `json:"jenkinsOrganizationPipeline"`
+	Name                        string `json:"name"`
 }
-type Artifacts struct {
+
+// GetSCMOrgRepo
+type TypeSCMOrgRepo struct {
```
Comment (zryfish) [inline pkg/models/devops/json.go]: TypeSCMOrgRepo ? not a good name

File: pkg/models/devops/devops.go (hunk 3 above)
Comment (soulseen) [reply pkg/models/devops/devops.go]: `Typepipeline` is Pipeline struct,and `RunPipeline` mean runs in a pipeline, such as https://github.com/jenkinsci/blueocean-plugin/tree/master/blueocean-rest#get-all-runs-in-a-pipeline.

Comment (soulseen) [reply pkg/models/devops/devops.go]: I will update this struct name.

File: pkg/apis/devops/v1alpha2/register.go (hunk 2 above)
Comment (soulseen) [reply pkg/apis/devops/v1alpha2/register.go]: ok,and I use `ok` for description message, other api use the default message is `ok`.

File: pkg/models/devops/json.go (hunk 4 above)
Comment (soulseen) [reply pkg/models/devops/json.go]: I change the name as `OrgRepo`.

File: pkg/models/devops/devops.go [hunk 5]
```diff
@@ -162,8 +162,101 @@ func Validate(scmId string, req *http.Request) ([]byte, error) {
 	return resBody, err
 }
 
-func GetOrgSCM(scmId string, req *http.Request) ([]interface{}, error) {
-	baseUrl := fmt.Sprintf(JenkinsUrl+GetOrgSCMUrl+req.URL.RawQuery, scmId)
+func GetSCMOrg(scmId string, req *http.Request) ([]interface{}, error) {
```
Comment (zryfish) [inline pkg/models/devops/devops.go]: `GetSCMOrg` `GetSCMOrgRepo` `StopPipeline` `ReplayPipeline` shares a lot of code in common, can we extract that part into one function?

File: pkg/models/devops/devops.go [hunk 6]
```diff
@@ -162,8 +162,101 @@ func Validate(scmId string, req *http.Request) ([]byte, error) {
...
+	return res, err
+}
+
+func ReplayPipeline(projectName, pipelineName, branchName, runId string, req *http.Request) (*ReplayPipe, error) {
+	baseUrl := fmt.Sprintf(JenkinsUrl+ReplayPipelineUrl+req.URL.RawQuery, projectName, pipelineName, branchName, runId)
+	log.Infof("Jenkins-url: " + baseUrl)
+
+	resBody, err := jenkinsClient(baseUrl, req)
+	if err != nil {
+		log.Error(err)
```
Comment (runzexia) [inline pkg/models/devops/devops.go]: https://travis-ci.org/kubesphere/kubesphere/builds/524867987#L499
run go fmt

**Pull request closed without merging**